File: extract/java/testdata/greeter/src/main/java/com/example/greeter/Greeter.java
package com.example.greeter;

/** Greeter greets by name. */
public class Greeter implements Speaker {
    private final String name;

    /** Greeter builds a Greeter for name. */
    public Greeter(String name) {
        this.name = name;
    }

    /** greet returns this Greeter's greeting. */
    public String greet() {
        return "hello, " + this.name;
    }
}

File: extract/rs/testdata/greeter/src/greeter.rs
//! Greeter greets by name.

/// Greeter greets by name.
pub struct Greeter {
    pub name: String,
}

impl Greeter {
    /// new builds a Greeter for name.
    pub fn new(name: &str) -> Greeter {
        Greeter { name: name.to_string() }
    }

    /// greet returns this Greeter's greeting.
    pub fn greet(&self) -> String {
        let mut out = String::from("hello, ");
        out.push_str(&self.name);
        out
    }
}
